cloud/linode/instances.go

```diff
@@ -79,12 +79,27 @@ func (e instanceNoIPAddressesError) Error() string {
 	return fmt.Sprintf("instance %d has no IP addresses", e.id)
 }
 
-func (i *instances) linodeByName(nodeName types.NodeName) (*linodego.Instance, error) {
+func (i *instances) linodeByIP(kNode *v1.Node) (*linodego.Instance, error) {
 	i.nodeCache.RLock()
 	defer i.nodeCache.RUnlock()
+	var kNodeAddresses []string
+	for _, address := range (kNode).Status.Addresses {
```

Comment:
Oh I forgot to mention the uneeded parentheses here. 
```suggestion
	for _, address := range kNode.Status.Addresses {
```